Framework/API/inc/MantidAPI/ComponentInfo.h

```diff
@@ -81,6 +81,8 @@ class MANTID_API_DLL ComponentInfo {
   bool hasParent(const size_t componentIndex) const;
   Kernel::V3D sourcePosition() const;
   Kernel::V3D samplePosition() const;
+  size_t sourceIndex() const;
+  size_t sampleIndex() const;
```

Comment:
Maybe drop the `Index` part of the name, as we did for all the other methods like `parent()`?